src/enumerate.rs

```diff
@@ -1 +1,202 @@
 //! State enumeration
+use crate::dfa;
+use crate::dfa::Dfa;
+use crate::dfa::State;
+use crate::eval;
+use crate::eval::RuntimeError;
+use crate::eval::Value;
+use crate::syntax::*;
+use std::collections::BTreeMap;
+
+type Env = BTreeMap<Id, Value>;
+
+/**
+ *
+ */
+pub fn enumerate(program: &Program, _input: &str) -> Result<Dfa<Symbol>, RuntimeError> {
+    // keep track of visited states and their environments
+    // let mut state_lookup: Map<State, Env> = Map::new();
+    let mut state_lookup: Map<State, Env> = Map::new();
+    // the transitions between states
```

Comment:
the names are descriptive enough, no need for any of the comments until the while loop below.